Core/include/Acts/Seeding/SeedFinderOrthogonal.ipp

```diff
@@ -0,0 +1,489 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/Seeding/SeedFilter.hpp"
+#include "Acts/Seeding/SeedFinderOrthogonalConfig.hpp"
+#include "Acts/Utilities/RangeXD.hpp"
+
+#include <cmath>
+#include <functional>
+#include <numeric>
+#include <type_traits>
+
+namespace Acts {
+enum Dim { DimPhi = 0, DimR = 1, DimZ = 2 };
+
+template <std::size_t N, typename external_spacepoint_t>
+RangeXD<N, float> validTupleOrthoRange(
+    const external_spacepoint_t &low,
+    const SeedFinderOrthogonalConfig<ActsExamples::SimSpacePoint> &config) {
+  float colMin = config.collisionRegionMin;
+  float colMax = config.collisionRegionMax;
+  float pL = low.phi();
+  float rL = low.radius();
+  float zL = low.z();
+
+  RangeXD<N, float> res;
+
+  res[DimPhi].shrinkMin(config.phiMin);
+  res[DimPhi].shrinkMax(config.phiMax);
+
+  res[DimR].shrinkMax(config.rMax);
+
+  res[DimZ].shrinkMin(config.zMin);
+  res[DimZ].shrinkMax(config.zMax);
+
+  res[DimR].shrinkMin(rL + config.deltaRMin);
+  res[DimR].shrinkMax(rL + config.deltaRMax);
+
+  float zMax = (res[DimR].max() / rL) * (zL - colMin) + colMin;
+  float zMin = colMax - (res[DimR].max() / rL) * (colMax - zL);
+
+  if (zL > colMin) {
+    res[DimZ].shrinkMax(zMax);
+  }
+
+  if (zL < colMax) {
+    res[DimZ].shrinkMin(zMin);
+  }
+
+  res[DimZ].shrinkMin(zL - config.cotThetaMax * (res[DimR].max() - rL));
+  res[DimZ].shrinkMax(zL + config.cotThetaMax * (res[DimR].max() - rL));
+
+  res[DimPhi].shrinkMin(pL - config.deltaPhiMax);
+  res[DimPhi].shrinkMax(pL + config.deltaPhiMax);
+
+  return res;
+}
+
+template <std::size_t N, typename internal_sp_t>
+RangeXD<N, float> validMidToLowOrthoRange(
+    const internal_sp_t &mid,
+    const SeedFinderOrthogonalConfig<ActsExamples::SimSpacePoint> &config) {
+  float pM = mid.phi();
+  float rM = mid.radius();
+
+  RangeXD<N, float> res;
+
+  res[DimPhi].shrinkMin(config.phiMin);
+  res[DimPhi].shrinkMax(config.phiMax);
+
+  res[DimR].shrinkMax(config.rMax);
+
+  res[DimZ].shrinkMin(config.zMin);
+  res[DimZ].shrinkMax(config.zMax);
+
+  res[DimR].shrinkMin(rM - config.deltaRMax);
+  res[DimR].shrinkMax(rM - config.deltaRMin);
+
+  float frac_r = res[DimR].min() / rM;
+
+  float zMin = (mid.z() - config.collisionRegionMin) * frac_r +
+               config.collisionRegionMin;
+  float zMax = (mid.z() - config.collisionRegionMax) * frac_r +
+               config.collisionRegionMax;
+
+  res[DimZ].shrinkMin(std::min(zMin, mid.z()));
+  res[DimZ].shrinkMax(std::max(zMax, mid.z()));
+
+  res[DimPhi].shrinkMin(pM - config.deltaPhiMax);
+  res[DimPhi].shrinkMax(pM + config.deltaPhiMax);
+
+  return res;
+}
+
+template <std::size_t N, typename internal_sp_t>
+bool validTupleOrtho(
```

Comment:
where is this declared? why is it never called?

Reply:
Removed.